src/macros.rs

```diff
@@ -138,6 +138,29 @@ macro_rules! define_from {
     }
 }
 
+macro_rules! define_from_smaller {
```

Comment:
I needed to be able to convert a 128-bit value to its 256-bit equivalent (E.g. `i64x2 -> i64x4`) in the `_mm256_broadcast*` functions. I created this macro to implement `From<T>` for this.

Reply:
Took a second look at this and realized I was misusing `simd_shuffle*`. No need for this now.